src/com/jalasoft/search_files/search/search.py

```diff
@@ -0,0 +1,180 @@
+        factory_asset.create_asset(path_absolute)
+        file_name, separator, extension = common_name.partition(".")
+        for item in factory_asset.get_list_actual_directories_and_files():
+            if which_search ==0:
+                if isinstance(item, File):
+                    if extension != "" :
+                        if file_name == item.get_file_name() and extension == item.get_extension():
+                            result.append(item)
+                    else:
+                        if file_name == item.get_file_name():
+                            result.append(item)
+                elif isinstance(item, Directory):
+                    if common_name == item.get_dir_name():
+                        result.append(item)
+            elif which_search == 2:
+                if isinstance(item, File):
+                    file_name, separator, extension = common_name.partition(".")
+                    if extension != "":
+                        if file_name == item.get_file_name() and extension == item.get_extension():
+                            result.append(item)
+                    else:
+                        if common_name == item.get_file_name():
+                            result.append(item)
+            elif which_search == 1:
+                if isinstance(item, Directory):
+                    if common_name == item.get_dir_name():
+                        result.append(item)
+
+        return result
+
+    def search_by_exact_common_extension(self, path_absolute, common_extension):
+        """
+                Args:
+                path_absolute (str): The first parameter.
+                common_extension (str): The second parameter.
+
+                Return:
+                Array [Asset]: The return directories and/or files list that have exact common extension
+
+                """
+        result= []
+        factory_asset = FactoryAsset()
+        factory_asset.create_asset(path_absolute)
+        file_name, separator, extension = common_extension.partition(".")
+        for item in factory_asset.get_list_actual_directories_and_files():
+            if isinstance(item, File):
+                if extension == item.get_extension():
+                    result.append(item)
+
+        return result
+
+    def search_files_by_less_than_a_size(self, path_absolute, size_on_megabites):
+        """
+                Args:
+                path_absolute (str): The first parameter.
+                size_on_megabites (int): The second parameter.
+
+                Return:
+                Array [Asset]: The return directories and/or files list that have less than a size
+
+                """
+        result= []
+        factory_asset = FactoryAsset()
+        factory_asset.create_asset(path_absolute)
+        for item in factory_asset.get_list_actual_directories_and_files():
+            if isinstance(item, File):
+                if  item.get_size()/(1024*1224) < size_on_megabites:
```

Comment:
Please remove extra space after of "if", just like below:
if item.get_size()/(1024*1224) < size_on_megabites: